src/spikeinterface/extractors/nwbextractors.py

```diff
@@ -1002,16 +1101,107 @@ def __init__(
-            spike_times_index_data=spike_times_index_data,
-            sampling_frequency=sampling_frequency,
-            t_start=self.t_start,
+        return units_ids, spike_times_data, spike_times_index_data
+
+    def extract_info_backend(
+        self, unit_table_path: str = None, samples_for_rate_estimation: int = 1000, cache: bool = False
+    ):
+        open_file = read_file_from_backend(
+            backend=self.backend,
+            file_path=self.file_path,
+            stream_mode=self.stream_mode,
+            cache=cache,
+            stream_cache_path=self.stream_cache_path,
+            storage_options=self.storage_options,
         )
-        self.add_sorting_segment(sorting_segment)
 
+        timestamps = None
+
+        if self.internal_sampling_frequency is None or self.t_start is None:
+            # defines the electrical series from where the sorting came from
+            # important to know the sampling_frequency
+            available_electrical_series = find_neurodata_type_from_backend(
+                open_file, neurodata_type="ElectricalSeries", backend=self.backend
+            )
+            if self.electrical_series_path is None:
+                if len(available_electrical_series) == 1:
+                    self.electrical_series_path = available_electrical_series[0]
+                else:
+                    raise ValueError(
+                        "Multiple ElectricalSeries found in the file. "
+                        "Please specify the 'electrical_series_path' argument:"
+                        f"Available options are: {available_electrical_series}."
+                    )
+            else:
+                if self.electrical_series_path not in available_electrical_series:
+                    raise ValueError(
+                        f"'{self.electrical_series_path}' not found in the file. "
+                        f"Available options are: {available_electrical_series}"
+                    )
+            electrical_series = open_file[self.electrical_series_path]
+
+            # Get sampling frequency
+            if "starting_time" in electrical_series.keys():
+                self.t_start = electrical_series["starting_time"][()]
+                self.internal_sampling_frequency = electrical_series["starting_time"].attrs["rate"]
+            elif "timestamps" in electrical_series.keys():
+                timestamps = electrical_series["timestamps"][:]
+                self.t_start = timestamps[0]
+                self.internal_sampling_frequency = 1.0 / np.median(np.diff(timestamps[:samples_for_rate_estimation]))
+
+        assert (
+            self.internal_sampling_frequency is not None
+        ), "Couldn't load sampling frequency. Please provide it with the 'sampling_frequency' argument"
+        assert (
+            self.t_start is not None
+        ), "Couldn't load a starting time for the sorting. Please provide it with the 't_start' argument"
+
+        if unit_table_path is None:
```

Comment:
This is a nice pattern. You add computation when an error to make the life of the user easier.